src/HDF5.jl

```diff
@@ -53,14 +53,14 @@ Base.cconvert(::Type{Ptr{T}}, ref::Reference) where {T<:Union{Reference,hobj_ref
 
 # Single character types
 # These are needed to safely handle VLEN objects
-abstract type CharType <: AbstractString end
+abstract type AbstractChar <: AbstractString end
```

Comment:
Base already has it's own `AbstractChar`, so I'm not sure shadowing it is a good idea.